docs/lofi-hifi-backends.md

```diff
@@ -0,0 +1,212 @@
+
+
+### ilab model train phased
+
+This command would take roughly the following arguments
+
+--optimizer=str (deepspeed, fsdp) describes the optimizer framework to use during training
+--device=str , accelerator to use cpu,cuda,rocm,mlx,mps
+--model-dir=path , dir where the model to be used in this phase is located
+--data-dir=path , dir where the data for this phase is located
+
+Note there is no Lora in this command and there is no quantization. The `ilab model train integrated` command will use the transformers library with pytorch. This is because those have awesome plugins for deepspeed and fsdp WITH lora and qlora. Those are absolute necessities for community usecase. However, they will be mostly unused in the "High Fidelity" usecase.
+
+### ilab data generate hifi
+
+This command would take something like the following arguments
+
+--num-samples=int
+--num-grounded-questions=int
+--num-gen-proc=int
+--num-util-proc=int (or is this for mixing?)
+
+
+### ilab data generate lofi
+
+This command would be the same as the existing `ilab generate`
+
+### ilab data mix
+
+This command would take something like the following arguments
+
+--num-util-proc=int
+--output-dir=str (defaults to generated/mixed)
+--knowledge-recipes=[]str (path to yaml)
+--skill-recipes=[]str (path to yaml)
+
+* Do we need an `ilab recipe` cmd? *
+
+
+## Workflows
+
+### ilab model train integrated
+
+A user on a Desktop with a consumer GPU (assume RTX 20/30 series) would run something like
+
+`ilab model train integrated --device=cuda --quantize --optimizer=deepspeed --num-epochs=5`
+
+This would 
+1. load the model in 4-bit-quantized form onto the GPUs vram
+2. setup the transformers trainer with this model, and with a hardcoded deepspeed config ilab would come with
+3. train for 5 epochs using fusedadam as the optim and deepspeed on top of that.
+4. give you a model in safetensors format (we cannot convert a quantized safetensor model)
+
+
+The big advantage here is more "fine tuned" training than currently exists in the CLI because of deepspeed (or fsdp). The user could even set this up for multi GPU or multi system support with future ilab enhancements.
+
+### ilab model train phased
+
+A user would run something like (assuming phase00 has run) on a GPU enabled server.
+
+`ilab model train phased --device=cuda --model-dir=./phase00/model --data-dir=./phase00/data`
+`ilab checkpoint evaluate ./phase05/checkpoints --output-dir=./phase10`
+`ilab model train phased --device=cuda --model-dir=./phase10/model --data-dir=./phase10/data`
+....
```

Comment:
This is the only place in the doc where `phase00`, `phase05`, and `phase10` are mentioned. I think these could use a more detailed definition of their meaning.